R/dup.R

```diff
@@ -5,14 +5,29 @@
 #'
 #' @inheritParams str_detect
 #' @param times Number of times to duplicate each string.
+#' @param sep String to insert between each duplicate.
 #' @return A character vector the same length as `string`/`times`.
 #' @export
 #' @examples
 #' fruit <- c("apple", "pear", "banana")
 #' str_dup(fruit, 2)
 #' str_dup(fruit, 1:3)
 #' str_c("ba", str_dup("na", 0:5))
-str_dup <- function(string, times) {
+str_dup <- function(string, times, sep = NULL) {
   vctrs::vec_size_common(string = string, times = times)
-  stri_dup(string, times)
+  if(is.null(sep)){
+    stri_dup(string, times)
+  } else {
+  # stri_dup does not currently support sep
+  check_string(sep)
+  lapply(seq_along(string), function(i) {
+    if (length(times) == 1) {
```

Comment:
I don't love how we have to condition on the length of `times` to handle vectorisation (i.e. when times is a scalar and string is a vector). Could we instead take the output from the `vctrs::vec_size_common` and use that?

Reply:
I have tried this but the result of vctrs::vec_size_common is the length of the string input, not the length of times

Reply:
Oh ooops needs to be `vec_recycle_common()`